advanced_functionality/multi_model_linear_learner_home_value/linear_learner_multi_model_endpoint_inf_pipeline.ipynb

```diff
@@ -0,0 +1,1302 @@
+    "1. [Exercise the inference pipeline - Get predictions from the different  linear learner models](#Exercise-inference-pipeline)\n",
+    "1. [Update Multi Model Endpoint with new models](#update-models)\n",
+    "1. [Fine grained control for model invocation using IAM conditional keys.](#Finegrain-control-invoke-models)\n",
+    "1. [Latency analysis of Linear Learner MME](#Latency-analysis)\n",
+    "1. [Clean up](#Clean-up)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "## Section 1 - Generate synthetic data for housing models <a id='Generate-synthetic-data-for-housing-models'></a>\n",
+    "\n",
+    "In this section, you will generate synthetic data that will be used to train the linear learner models.  The data generated consists of 6 numerical features - the year the house was built in, house size in square feet, number of bedrooms, number of bathroom, the lot size and number of garages and two categorial features - deck and front_porch.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Import necessary libraries\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import json\n",
+    "import datetime\n",
+    "import time\n",
+    "import boto3\n",
+    "import sagemaker\n",
+    "import os\n",
+    "\n",
+    "from time import gmtime, strftime\n",
+    "from random import choice\n",
+    "\n",
+    "from sagemaker import get_execution_role\n",
+    "from sagemaker.predictor import csv_serializer\n",
+    "\n",
+    "from sagemaker.multidatamodel import MULTI_MODEL_CONTAINER_MODE\n",
+    "from sagemaker.multidatamodel import MultiDataModel\n",
+    "\n",
+    "from sklearn.model_selection import train_test_split"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "NUM_HOUSES_PER_LOCATION = 1000\n",
+    "LOCATIONS  = ['NewYork_NY',    'LosAngeles_CA',   'Chicago_IL',    'Houston_TX',   'Dallas_TX',\n",
+    "              'Phoenix_AZ',    'Philadelphia_PA', 'SanAntonio_TX', 'SanDiego_CA',  'SanFrancisco_CA']\n",
+    "PARALLEL_TRAINING_JOBS = 4 # len(LOCATIONS) Reduce or increase according to your AWS account limitations\n",
+    "MAX_YEAR = 2019"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Function to generate price based on features of the house\n",
+    "def gen_price(house):\n",
+    "    _base_price = int(house['SQUARE_FEET'] * 150)\n",
+    "    if (house['FRONT_PORCH'] == 'y'):\n",
+    "        _garage = 1\n",
+    "    else:\n",
+    "        _garage = 0;\n",
+    "        \n",
+    "    if (house['FRONT_PORCH'] == 'y'):\n",
+    "        _front_porch = 1\n",
+    "    else:\n",
+    "        _front_porch = 0;\n",
```

Comment:
Leaving this as is, because I need to showcase the preprocessing as part of the inference pipeline.  Preprocessing is to one hot encode the categorical values.